sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_ADAPTER
+#define AZ_CURL_ADAPTER
+
+#include <stdio.h>
+#include <curl/curl.h>
+#include <az_span.h>
+
```

Comment:
Please include az_*.h first. Then the rest, after an empty line (otherwise clang-format will sort all them as a singe block). Then, after another empty line the last include should be _az_cfg_prefix.h. And _az_cfg_suffix in the end - see other header files. In .c files, you want _az_cfg.h instead of _az_cfg_prefix, and no _az_cfg_suffix.